pkg/types/types.go

```diff
@@ -1267,6 +1267,46 @@ func NewImageAttributes(alt, width, height interface{}, otherattrs []interface{}
 	return result, nil
 }
 
+// ------------------------------------------
+// Icons
+// ------------------------------------------
+
+type Icon struct {
+	Class      string
+	Attributes Attributes
+}
+
+// NewImageAttributes returns a map of image attributes, some of which have implicit keys (`alt`, `width` and `height`)
+func NewIconAttributes(size interface{}, others []interface{}) (Attributes, error) {
+	var result Attributes
+
+	if alt, ok := size.(string); ok {
+		if sizeStr := Apply(alt, strings.TrimSpace); sizeStr != "" {
+			result = result.Set(AttrIconSize, sizeStr)
+		}
+	}
+	for _, otherAttr := range others {
+		if otherAttr, ok := otherAttr.(Attributes); ok {
+			for k, v := range otherAttr {
+				result = result.Set(k, v)
+				if k == AttrID {
+					// mark custom_id flag to `true`
+					result = result.Set(AttrCustomID, true)
+				}
+			}
+		}
+	}
+	return result, nil
+}
+
+// NewInlineIcon initializes a new `InlineIcon`
+func NewInlineIcon(class string, attributes Attributes) (Icon, error) {
```

Comment:
You could probably rename this func to `NewIcon`

Reply:
Reasonable idea.  At one point I had a separate type for InlineIcon.  I guess I should clean this up too.